spec/scenic/schema_dumper_spec.rb

```diff
@@ -36,4 +36,50 @@ class Search < ActiveRecord::Base; end
       Search.connection.drop_view :'scenic.searches'
     end
   end
+
+  context "with dependent views " do
+    it "dumps create_view sorted alphabetically" do
+      views = {
+        "view1": "SELECT 'needle1'::text AS haystack",
+        "view2": "SELECT 'needle2'::text AS haystack",
+        "view1_1": "SELECT * FROM view1",
+        "view1_2": "SELECT * FROM view1",
+        "a_view1_1_1": "SELECT * FROM view1_1",
+        "c_view1_and_1_1": "SELECT * FROM view1_1 UNION SELECT * FROM view1",
+        "z_view1_1_2": "SELECT * FROM view1_1",
+      }
+
+      expected_order = %w(
+        view1
+        view1_1
+        a_view1_1_1
+        view1_2
+        view2
+        z_view1_1_2
+      )
+
+      views.each do |name, definition|
+        Search.connection.create_view(name, sql_definition: definition)
+      end
+
+      stream = StringIO.new
+
+      ActiveRecord::SchemaDumper.dump(Search.connection, stream)
+
+      output = stream.string
+
+      views.each do |name, _|
+        expect(output).to include "create_view :#{name},"
+      end
+
+      create_views = output.lines.grep(/create_view :/)
+      order = create_views.map { |line| line.scan(/:([a-z_0-9]+),/).first.first }
+
+      expect(order).to eq(expected_order)
```

Comment:
@mcmire isn't this testing the behavior you describe [in your comment](https://github.com/scenic-views/scenic/pull/191#issuecomment-498364427)?

Reply:
It is, that's true, although I'm struggling to see how, since all this PR seems to do is change how views are being sorted when they're spit out (other than the materialized view stuff, but it seems like that's separate?), and it changes it so it's no longer an alphabetical sort but a sort on `oid`. I'm not 100% certain what that is, I'm guessing it corresponds to the order in which the view was added initially, but there are some question marks here for sure. I'll have to dive into this further.